File: ContextExecutorJSModule.js
var BatchedBridge = require('BatchedBridge');

export default function ContextExecute(onRun) {
  this.init = onRun => {
    this.onRun = onRun;
    BatchedBridge.registerCallableModule('ContextExecutorJSModule', this);
  }

  this.execute = (op, data) => {
    this.onRun(op, data);
  }

  this.init(onRun);
  return this;
};

File: android/src/main/java/com/sreejithr/contextexecute/ContextExecutorJSModule.java
package com.sreejithr.contextexecute;

import com.facebook.react.bridge.JavaScriptModule;
import com.facebook.react.bridge.WritableMap;

/**
 * Created by sreejith on 20/02/17.
 */

public interface ContextExecutorJSModule extends JavaScriptModule {
    void execute(String op, WritableMap data);
}